Peer dependency error with React 19.x

### Package

`graphql-hooks`

### Environment

- `graphql-hooks` 8.0.1:
- `react` 19.x:
- Browser: Chrome

### Description

```
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: keva-frontend@0.1.0
npm error Found: react@19.0.0
npm error node_modules/react
npm error   react@"19.0.0" from the root project
npm error
npm error Could not resolve dependency:
npm error peer react@"^17.0.0 || ^18.0.0" from graphql-hooks@8.0.1
npm error node_modules/graphql-hooks
npm error   graphql-hooks@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error C:\Users\mickeymond\AppData\Local\npm-cache\_logs\2024-12-17T19_31_08_440Z-eresolve-report.txt
npm error A complete log of this run can be found in: C:\Users\mickeymond\AppData\Local\npm-cache\_logs\2024-12-17T19_31_08_440Z-debug-0.log
```

### How to reproduce

- Create a new nextjs project that used React 19.x
- Try installing graphql-hooks package: `npm i graphql-hooks`

### Suggested solution (optional)

Update the peer-dependencies to support React 19.x

--- a/packages/graphql-hooks/package.json
+++ b/packages/graphql-hooks/package.json
@@ -33,7 +33,7 @@
   "author": "Brian Mullan <bmullan91@gmail.com>",
   "license": "Apache-2.0",
   "peerDependencies": {
-    "react": "^17.0.0 || ^18.0.0"
+    "react": "^17.0.0 || ^18.0.0 || ^19.0.0"
   },
   "dependencies": {
     "@0no-co/graphql.web": "^1.0.7",

